lib/azure/armrest/storage_account_service.rb

```diff
@@ -146,7 +197,11 @@ def list_private_images(group = configuration.resource_group)
 
         list(group).each do |lstorage_account|
           threads << Thread.new(lstorage_account) do |storage_account|
-            key = list_account_keys(storage_account.name, group).fetch('key1')
+            if Time.parse(api_version).utc >= Time.parse('2016-01-01').utc
```

Comment:
`Time.parse(api_version).utc >= Time.parse('2016-01-01').utc` appears multiple times in this class. Please extract it to a dedicated method with meaningful name.